.github/workflows/notebooks.yaml

```diff
@@ -0,0 +1,47 @@
+name: Notebooks
+
+on:
+  # Trigger the workflow on push or pull request, but only on dev branch
+  push:
+    branches:
+      - revamp-2025
+  pull_request:
+    branches:
+      - revamp-2025
+
+permissions:
+  contents: read  # to fetch code
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  documentation:
+    name: Set up Python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8  # v5.0.0
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c  # v6.0.0
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[dev]
+        pip install ipykernel pytest-notebook
+        pip freeze
+    - name: Register Python3 kernel
+      run: |
+        python -m ipykernel install --user --name "python3" --display-name "Python 3"
+    - name: Turn on test mode and test notebooks
+      run: |
```

Comment:
We can use the `env:` block above `run:` to set environment variables. There's an example of this here: https://github.com/jax-ml/jax/blob/41bb2d7ebb18d9e67936db82d45a01eab2d69375/.github/workflows/ci-build.yaml#L80